lib/utils/aws/fips_disabled.go

```diff
@@ -0,0 +1,42 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package aws
+
+import (
+	"os"
+	"strconv"
+)
+
+// IsFIPSDisabledByEnv returns true if the TELEPORT_UNSTABLE_DISABLE_AWS_FIPS
+// environment variable is set.
+//
+// Either "yes" or a "truthy" value (as defined by [strconv.ParseBool]) are
+// considered true.
+//
+// Prefer using specific functions, such as those in the
+// lib/utils/aws/stsutils or lib/utils/aws/dynamodbutils packages.
+func IsFIPSDisabledByEnv() bool {
+	const envVar = "TELEPORT_UNSTABLE_DISABLE_AWS_FIPS"
```

Comment:
I'm not sure about this env var name and how it will handle future changes

Let's say another customer comes to us and they really need to use the non-FIPS endpoint for S3. Do we make this same env var disable FIPS for S3? But would this be a breaking change for anyone already using this env var to disable FIPS endpoints only for dynamo and sts, but they do want to use FIPS for S3, or at least prevent the possibility of accidentally use non-FIPS for S3?

Or are we just okay with that and that's why we included UNSTABLE in the name. I guess the alternative is to use a separate scoped env var for each AWS service where we need this option

Reply:
I thought about that too, but splitting the variables seemed awkward in terms of "UX" (they'd have to set multiple _DISABLE variables) and, at the same time, too much product commitment: if we split it means that, at some level, we are committed to per-service toggles.

I think we have, broadly, the following approaches:

1. Go with a single variable now, potentially split it later if it comes up (so we'd have "TELEPORT_UNSTABLE_DISABLE_AWS_FIPS" and "TELEPORT_UNSTABLE_DISABLE_AWS_S3_FIPS", in your example)
2. Go with a single variable now, don't split if we add more services
3. Split now


(2) is basically what I was asked to do, plus it seems to match the (admittedly few) reports I saw.

Reply:
FWIW, I think the "ideal" solution would be to selectively fallback if there's no FIPS endpoint for a given service in that region, maybe with the addition of a "strict" toggle that panics if some FIPS endpoint is lacking.

That is a nicer, but more evolved change than I have here.

There's also a question of FIPS binary vs the --fips flag and a possible simplification of it all.

Reply:
(Oh, also you can disable FIPS for S3 in the audit_sessions_uri.)